linode/manager
# change: [M3-8000] - GPUv2 Plan Divider & PlanSelection cleanup

## Description 📝
This PR has two separate goals:
- continue the consolidation of the Linode Create & K8 Create SelectionPlan components
- implement the new plan divider for GPU plans

There may be more follow up as new features build on our Plan Selection interfaces, but this refactor will help much in reducing the barrier to doing so.

## Changes  🔄
- Improve PlansPanel (and children) APIs
  - props
  - naming conventions
  - reusability
- Improve plans selection accessibility
- Add new reusable logic to enable dividing plans into separate tables
- Add new `gpuv2` feature flag to control Divider for GPU plans
- Improve `ExtendedTypes` to include props tied to plan disabled state
- Consolidate Disabled Tooltip
  - New Component (we have two tooltips previously)
  - Copy (via util)
  - Placement (both tables and cards)
- Update and increase selection-plan cypress test

## Target release date 🗓️
**5/13/2024**

## Preview 📷
The major visual change resulting from this PR is the Table divider feature shown below

| Desktop  | Mobile   |
| ------- | ------- |
| ![Screen Shot 2024-05-01 at 13 52 12](https://github.com/linode/manager/assets/130582365/05666358-a152-48a9-aa17-2d782fefe92b) |  ![Screen Shot 2024-05-01 at 13 55 15](https://github.com/linode/manager/assets/130582365/93b15744-3697-4ae8-98c5-e7f73fe1dcce) |

Another changes include the Card chip opacity (opacity for copy but not "Limited Development Availability" chip

| Before  | After   |
| ------- | ------- |
| ![Screen Shot 2024-05-01 at 10 55 25](https://github.com/linode/manager/assets/130582365/93116227-5f39-4101-9faa-22c07b063a87) | ![Screen Shot 2024-05-01 at 10 54 58](https://github.com/linode/manager/assets/130582365/9e1fb115-7e3d-430f-995d-304f69ea7fa9) |

## How to test 🧪

### Verification steps
ℹ️ Using ALPHA (the new "Ada" plans are only returned there at the moment)
ℹ️ The new `gpuv2` flag is `ON` in development only and I didn't add it to the dev tools as it did not seem the switching feature was important here.

- In the Linode create flow, select "Newark, NJ" region
  - Select the GPU tab and confirm display and functionality of the new split tables
  - Confirm other plan tabs are not affected
- In the Kubernetes Create Flow, confirm no regression
- In the Database Create & Resize flows, confirm no regression

## As an Author I have considered 🤔

*Check all that apply*
